orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -251,14 +251,18 @@ def forward(ctx, *inputs, **kwargs):
                 return user_outputs
 
             @staticmethod
-            def backward(ctx, *grad_output):
+            def backward(ctx, *grad_outputs):
                 '''Performs backward pass based on grad wrt module output
                 '''
 
                 # Use IO binding
                 # Push user output grads to ONNX backend.
                 backward_grad_output_ortvalue = []
-                for grad_output in grad_output[:len(self._onnx_graphs_info.backward_output_grad_names)]:
+                for grad_output in grad_outputs[:len(self._onnx_graphs_info.backward_output_grad_names)]:
+                    # Force torch tensors to be contiguous before converting into OrtValue
+                    if not grad_output.is_contiguous():
+                        grad_output = grad_output.contiguous()
```

Comment:
Looks like the variable scope in Python is different from other languages, this grad_output tensor is not released when going out of the for loop, otherwise, the data at grad_output.data_ptr() will be undefined when it goes to ORT.

Reply:
Do you expect perf overhead with this? A possible memory call could take place here

Reply:
@iK1D 
I think grad_output is taking a reference to grad_outputs, so the changes persist over the loop scope.

Reply:
@thiagocrepaldi 
Yes, there will be perf overhead if the tensor happens to be not contiguous, and a copy is inevitable, coz outvalue only support contiguous format. 

The hope is that pytorch doesn't have many cases where a non-contiguous tenser is returned. 
We caught this bug, because the gradient of .sum returns a sparse tensor.

Reply:
Yeah, some copy is inevitable. I was just double checking whether we were on the same page :)
One example for such non contiguousness is transposing tensors, but there might be other cases out there.

Reply:
@SherlockNoMad If we don't call the contiguous(), grad_output is taking reference to grad_outputs, but when we call it, there should be a new tensor created inside the for loop. But it works. :-)